When I run the test locally I see some issues that seem related. We must make sure the mocking is working correctly.

```
FAILED tests/components/alexa/test_state_report.py::test_proactive_mode_filter_states - AssertionError: No mock registered for POST https://api.amazonalexa.com/v3/events None

Results (15.60s):
     244 passed
       1 error
ERROR:homeassistant:Error doing job: Task exception was never retrieved:   File "/usr/local/py-utils/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/config/__init__.py", line 198, in console_main
    code = main()
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/config/__init__.py", line 175, in main
    ret: Union[ExitCode, int] = config.hook.pytest_cmdline_main(
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/main.py", line 320, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/main.py", line 273, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/main.py", line 327, in _main
    config.hook.pytest_runtestloop(session=session)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/main.py", line 352, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 115, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 134, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 223, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 262, in call_runtest_hook
    return CallInfo.from_call(
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 342, in from_call
    result: Optional[TResult] = func()
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 263, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 170, in pytest_runtest_call
    item.runtest()
  File "/usr/local/lib/python3.12/site-packages/pytest_asyncio/plugin.py", line 436, in runtest
    super().runtest()
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/python.py", line 1831, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/python.py", line 194, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/usr/local/lib/python3.12/site-packages/pytest_asyncio/plugin.py", line 897, in inner
    _loop.run_until_complete(task)
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 672, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 639, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1977, in _run_once
    handle._run()
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/workspaces/core/tests/components/alexa/test_state_report.py", line 748, in test_proactive_mode_filter_states
    hass.states.async_set(
  File "/workspaces/core/homeassistant/core.py", line 2196, in async_set
    self._bus._async_fire(  # pylint: disable=protected-access
  File "/workspaces/core/homeassistant/core.py", line 1456, in _async_fire
    self._hass.async_run_hass_job(job, event)
  File "/workspaces/core/homeassistant/core.py", line 843, in async_run_hass_job
    return self.async_add_hass_job(
  File "/workspaces/core/homeassistant/core.py", line 683, in async_add_hass_job
    task = create_eager_task(
  File "/workspaces/core/homeassistant/util/async_.py", line 27, in create_eager_task
    return Task(
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/components/alexa/state_report.py", line 337, in _async_entity_state_listener
    await async_send_changereport_message(
  File "/workspaces/core/homeassistant/components/alexa/state_report.py", line 390, in async_send_changereport_message
    response = await session.post(
               ^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/tests/test_util/aiohttp.py", line 157, in match_request
    raise AssertionError(f"No mock registered for {method.upper()} {url} {params}")
AssertionError: No mock registered for POST https://api.amazonalexa.com/v3/events None
ERROR:homeassistant:Error doing job: Task exception was never retrieved:   File "/usr/local/py-utils/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/config/__init__.py", line 198, in console_main
    code = main()
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/config/__init__.py", line 175, in main
    ret: Union[ExitCode, int] = config.hook.pytest_cmdline_main(
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/main.py", line 320, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/main.py", line 273, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/main.py", line 327, in _main
    config.hook.pytest_runtestloop(session=session)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/main.py", line 352, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 115, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 134, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 223, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 262, in call_runtest_hook
    return CallInfo.from_call(
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 342, in from_call
    result: Optional[TResult] = func()
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 263, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/runner.py", line 170, in pytest_runtest_call
    item.runtest()
  File "/usr/local/lib/python3.12/site-packages/pytest_asyncio/plugin.py", line 436, in runtest
    super().runtest()
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/python.py", line 1831, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/py-utils/venvs/pytest/lib/python3.12/site-packages/_pytest/python.py", line 194, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/usr/local/lib/python3.12/site-packages/pytest_asyncio/plugin.py", line 897, in inner
    _loop.run_until_complete(task)
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 672, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 639, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1977, in _run_once
    handle._run()
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/workspaces/core/tests/components/alexa/test_state_report.py", line 761, in test_proactive_mode_filter_states
    hass.states.async_set(
  File "/workspaces/core/homeassistant/core.py", line 2196, in async_set
    self._bus._async_fire(  # pylint: disable=protected-access
  File "/workspaces/core/homeassistant/core.py", line 1456, in _async_fire
    self._hass.async_run_hass_job(job, event)
  File "/workspaces/core/homeassistant/core.py", line 843, in async_run_hass_job
    return self.async_add_hass_job(
  File "/workspaces/core/homeassistant/core.py", line 683, in async_add_hass_job
    task = create_eager_task(
  File "/workspaces/core/homeassistant/util/async_.py", line 27, in create_eager_task
    return Task(
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/components/alexa/state_report.py", line 337, in _async_entity_state_listener
    await async_send_changereport_message(
  File "/workspaces/core/homeassistant/components/alexa/state_report.py", line 390, in async_send_changereport_message
    response = await session.post(
               ^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/tests/test_util/aiohttp.py", line 157, in match_request
    raise AssertionError(f"No mock registered for {method.upper()} {url} {params}")
AssertionError: No mock registered for POST https://api.amazonalexa.com/v3/events None
```

--- a/tests/components/alexa/test_state_report.py
+++ b/tests/components/alexa/test_state_report.py
@@ -731,39 +731,39 @@ async def test_proactive_mode_filter_states(
     assert len(aioclient_mock.mock_calls) == 0
 
     # hass not running should not report
+    current_state = hass.state
+    hass.set_state(core.CoreState.stopping)
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
     hass.states.async_set(
         "binary_sensor.test_contact",
         "off",
         {"friendly_name": "Test Contact Sensor", "device_class": "door"},
     )
 
-    current_state = hass.state
-    hass.set_state(core.CoreState.stopping)
-    await hass.async_block_till_done()
-    await hass.async_block_till_done()
     hass.set_state(current_state)
     assert len(aioclient_mock.mock_calls) == 0
 
     # unsupported entity should not report
-    hass.states.async_set(
-        "binary_sensor.test_contact",
-        "on",
-        {"friendly_name": "Test Contact Sensor", "device_class": "door"},
-    )
     with patch.dict(
         "homeassistant.components.alexa.state_report.ENTITY_ADAPTERS", {}, clear=True
     ):
+        hass.states.async_set(
+            "binary_sensor.test_contact",
+            "on",
+            {"friendly_name": "Test Contact Sensor", "device_class": "door"},
+        )
         await hass.async_block_till_done()
         await hass.async_block_till_done()
     assert len(aioclient_mock.mock_calls) == 0
 
     # Not exposed by config should not report
-    hass.states.async_set(
-        "binary_sensor.test_contact",
-        "off",
-        {"friendly_name": "Test Contact Sensor", "device_class": "door"},
-    )
     with patch.object(config, "should_expose", return_value=False):
+        hass.states.async_set(
+            "binary_sensor.test_contact",
+            "off",
+            {"friendly_name": "Test Contact Sensor", "device_class": "door"},
+        )
         await hass.async_block_till_done()
         await hass.async_block_till_done()
     assert len(aioclient_mock.mock_calls) == 0

